Applications/Utils/MeshGeoTools/ComputeSurfaceNodeIDsInPolygonalRegion.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * @file ComputeSurfaceNodeIDsInPolygonalRegion.cpp
+ * @date 2014-05-20
+ * @brief Computes mesh node ids of mesh nodes within a polygonal region, that resides on the surface.
+ *
+ * @copyright
+ * Copyright (c) 2014, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+#include "tclap/CmdLine.h"
+#include "logog/include/logog.hpp"
+
+#include "BaseLib/LogogSimpleFormatter.h"
+#include "BaseLib/StringTools.h"
+#include "BaseLib/FileTools.h"
+
+// FileIO
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/readGeometryFromFile.h"
+
+// GeoLib
+#include "GeoLib/GEOObjects.h"
+#include "GeoLib/Polygon.h"
+
+// MathLib
+#include "Vector3.h"
+
+// MeshLib
+#include "Mesh.h"
+#include "Node.h"
+#include "MeshSurfaceExtraction.h"
+
+void writeToFile(std::string const& id_area_fname, std::string const& csv_fname,
+	std::vector<std::pair<std::size_t, double>> const& ids_and_areas,
+	std::vector<MeshLib::Node*> const& mesh_nodes)
+{
+	std::ofstream ids_and_area_out(id_area_fname);
+	std::ofstream csv_out(csv_fname);
+	ids_and_area_out << ids_and_areas[0].first << " " << ids_and_areas[0].second;
+	csv_out << "ID x y z area node_id\n"; // CSV header
+	csv_out << 0 << " " << *mesh_nodes[ids_and_areas[0].first]
+			<< ids_and_areas[0].second << " " << ids_and_areas[0].first;
+	for (std::size_t k(0); k<ids_and_areas.size(); k++) {
+		ids_and_area_out << "\n" << std::setprecision(20)
+			<< ids_and_areas[k].first << " " << ids_and_areas[k].second;
+		csv_out << "\n" << k << " " << std::setprecision(20)
+			<< *mesh_nodes[ids_and_areas[k].first]
+			<< ids_and_areas[k].second << " " << ids_and_areas[k].first;
+	}
+	ids_and_area_out.close();
+	csv_out.close();
+}
+
+int main (int argc, char* argv[])
+{
+	LOGOG_INITIALIZE();
+	logog::Cout* logog_cout (new logog::Cout);
+	BaseLib::LogogSimpleFormatter *custom_format (new BaseLib::LogogSimpleFormatter);
+	logog_cout->SetFormatter(*custom_format);
```

Comment:
As an experiment, could you try to use `LogogSetup.h` for initialization (as in `ogs.cpp`)?